src/EditorFeatures/CSharpTest/UsePatternCombinators/CSharpUsePatternCombinatorsDiagnosticAnalyzerTests.cs

```diff
@@ -477,6 +479,85 @@ void M(char c)
         }
     }
 }
+");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUsePatternCombinators)]
+        [WorkItem(57199, "https://github.com/dotnet/roslyn/issues/57199")]
+        public async Task TestMissingInNonConvertibleTypePattern1()
+        {
+            await TestMissingAsync(
+@"
+static class C
+{
+    public struct S1 : I { }
+    public struct S2 : I { }
+    public interface I { }
+}
+
+class Test<T>
+{
+    public readonly T C;
+    bool P => [|C is C.S1 || C is C.S2|];
+}
+");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUsePatternCombinators)]
+        [WorkItem(57199, "https://github.com/dotnet/roslyn/issues/57199")]
+        public async Task TestMissingInNonConvertibleTypePattern2()
+        {
+            await TestMissingAsync(
+@"
+        private class Goo
```

Comment:
The most outer class is private. This is a compile error. Consider making public